### ISSUE TITLE
Exhaustive switch return type does not work when wrapped in a try-catch.

### ISSUE THREAD

**Issue by trusktr** [microsoft/TypeScript#32905]
**TypeScript Version:**  3.4.5

**Search Terms:**

Exhaustive switch return type does not work when wrapped in a try-catch

**Code**

```ts
enum Foo {
  One,
  Two,
  Three
}

function test(type: Foo): number {
  switch (type) {
    case Foo.One:
      return 0;
    case Foo.Two:
      return 0;
    case Foo.Three:
      return 0;
  }
}

function test2(type: Foo): number { // ERROR
  try {
    switch (type) {
      case Foo.One:
        return 0;
      case Foo.Two:
        return 0;
      case Foo.Three:
        return 0;
    }
  } catch (e) {
      throw new Error('some error')
  }
}
```

[playground](http://www.typescriptlang.org/play/#code/KYOwrgtgBAYg9nKBvAUFKB5EwA0aoAqA7nHugQBYBOwwKAviigGZggDGALgJZwhSdgAZ04AKTgE8ADsABcsBAEp54CACNgVZPiFFundhSjjpwRdvTp2AQyHAFcAHRY5+S1BqcwVfgAYA3G5QNnYOjsRwskHont5+ge4h9vBOlDSu7jHAXj5QAfiMjCxsXLz8giIATCYy8inKUKoaWqjonFQSFpa6+obGkjLmrZlJYS5RmZaxufmToynhJBOTHtlxeQkjtskI4dS0y5PT8UGM6PTB1gZGomZd7pzUcESNwC8AolRUcFSiAORCOAQeyab5UP6KAoMFBAA)

**Expected behavior:**

It should still work (because clearly in this case it can only return those values, or throw an error).

**Actual behavior:**

Error

**Comment by RyanCavanaugh** [comment]
All of the function-always-returning-from-exhaustive-switch forms are special-cased; this one seems like a good additional special case

**Comment by jack-williams** [comment]
Fix here: #32952.

**Comment by jack-williams** [comment]
@ahejlsberg fixed this in #32695.

**Comment by trusktr** [comment]
Looks like fixed, tested in playground. Thanks!
